docs/src/verification.md

```diff
@@ -0,0 +1,164 @@
+# Verification
+
+## General Overview
+
+When comparing our `compiled_model` with the `framework_model` (e.g., `PyTorch`), we aim to verify whether the output from the `compiled_model` is sufficiently similar to the output from the `framework_model` (where required degree of similarity is configurable).
+
+So generally we want to perform the following steps:
+1. Create a framework model
+2. Run a forward pass through the framework model
+3. Compile the framework model using `Forge`
+4. Run a forward pass through the compiled model
+5. Compare the outputs.
+
+Most of the above steps `verify()` function does for us:
+- Handles forward passes for both framework and compiled models
+- Compares results using a combination of comparison methods
+- Supports customization through the `VerifyConfig` class.
+
+**Example of usage**
+```python
+def test_add():
+
+	class Add(nn.Module):
+		def __init__(self):
+			super().__init__()
+
+		def forward(self, a, b):
+			return a + b
+
+
+	inputs = [torch.rand(2, 32, 32), torch.rand(2, 32, 32)]
+
+	framework_model = Add()
+	compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+
+	verify(inputs, framework_model, compiled_model)
+```
+
+**Notes:**
+- If you only want to compile model and perform forward pass without comparing outputs you can just:
+```python
+def test_add():
+
+	class Add(nn.Module):
+		def __init__(self):
+			super().__init__()
+
+		def forward(self, a, b):
+			return a + b
+
+
+	inputs = [torch.rand(2, 32, 32), torch.rand(2, 32, 32)]
+
+	framework_model = Add()
+	compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+
+	fw_out = framework_model(*inputs)
+	co_out = compiled_model(*inputs)
+```
+
+## Verify Config Overview
+
+If `VerifyConfig` isn't passed as a param, default one will be used. Currently through `VerifyConfig` you can disable/enable:
+| Feature                           | Enabled (default) |
+|-----------------------------------|:-----------------:|
+| Verification as a method          | `True`            |
+| Output size check                 | `True`            |
```

Comment:
what is the output size?

Reply:
I've changed table a bit, hopefully it's clearer now.

Reply:
what is the output length? 😄 is it the number of output tensors?

Reply:
hahah yes (I will put number of output tensors instead)